lib/Controller/ChatController.php

```diff
@@ -332,6 +335,262 @@ public function sendMessage(string $message, string $actorDisplayName = '', stri
 		return $this->parseCommentToResponse($comment, $parentMessage);
 	}
 
+	/**
+	 * Get all scheduled nessages of a given room and participant
+	 *
+	 * The author and timestamp are automatically set to the current user
+	 * and time.
+	 *
+	 * Required capability: `scheduled-messages`
+	 *
+	 * @return DataResponse<Http::STATUS_OK, list<TalkScheduledMessage>, array{}>|DataResponse<Http::STATUS_BAD_REQUEST|Http::STATUS_NOT_FOUND, array{error: 'actor'|'message'}, array{}>
+	 *
+	 * 200: All scheduled messages for this room and participant
+	 * 400: Could not get scheduled messages
+	 * 404: Actor not found
+	 */
+	#[NoAdminRequired]
+	#[RequireModeratorOrNoLobby]
+	#[RequireParticipant]
+	#[RequirePermission(permission: RequirePermission::CHAT)]
+	#[RequireReadWriteConversation]
+	#[ApiRoute(verb: 'GET', url: '/api/{apiVersion}/chat/{token}/schedule', requirements: [
+		'apiVersion' => '(v1)',
+		'token' => '[a-z0-9]{4,30}',
+	])]
+	public function scheduledMessages(): DataResponse {
+		if ($this->participant->isSelfJoinedOrGuest()) {
+			return new DataResponse(['error' => 'actor'], Http::STATUS_NOT_FOUND);
+		}
+
+		try {
+			$scheduledMessages = $this->scheduledMessageManager->getMessages(
+				$this->room,
+				$this->participant,
+			);
+		} catch (\Exception $e) {
+			$this->logger->warning($e->getMessage());
+			return new DataResponse(['error' => 'message'], Http::STATUS_BAD_REQUEST);
+		}
+
+		return new DataResponse($scheduledMessages, Http::STATUS_OK);
+	}
+
+	/**
+	 * Schedules the sending of a new chat message to the given room
+	 *
+	 * The author and timestamp are automatically set to the current user
+	 * and time.
+	 *
+	 * Required capability: `scheduled-messages`
+	 *
+	 * @param string $message The message to send
+	 * @param int $sendAt When to send the scheduled message
+	 * @param int $replyTo Parent id which this scheduled message is a reply to
+	 * @psalm-param non-negative-int $replyTo
+	 * @param bool $silent If sent silent the scheduled message will not create any notifications when sent
+	 * @param string $threadTitle Only supported when not replying, when given will create a thread (requires `threads` capability)
+	 * @param int $threadId Thread id without quoting a specific message (requires `threads` capability)
+	 * @return DataResponse<Http::STATUS_CREATED, TalkScheduledMessage, array{}>|DataResponse<Http::STATUS_BAD_REQUEST|Http::STATUS_NOT_FOUND|Http::STATUS_REQUEST_ENTITY_TOO_LARGE, array{error: string}, array{}>
```

Comment:
Let's be more specific

```suggestion
	 * @return DataResponse<Http::STATUS_CREATED, TalkScheduledMessage, array{}>|DataResponse<Http::STATUS_BAD_REQUEST, array{error: 'message'|'reply-to'|'send-at'}, array{}>|DataResponse<Http::STATUS_REQUEST_ENTITY_TOO_LARGE, array{error: 'message'}, array{}>|DataResponse<Http::STATUS_NOT_FOUND, array{error: 'actor'}, array{}>
```